_topics/en/transaction-bloom-filtering.md

```diff
@@ -0,0 +1,115 @@
+---
+title: Transaction bloom filtering
+
+## Optional.  An entry will be added to the topics index for each alias
+aliases:
+  - BIP37
+  - Bloom filters
+
+## Required.  At least one category to which this topic belongs.  See
+## schema for options
+categories:
+  - Lightweight Client Support
+  - Privacy Problems
+  - P2P Network Protocol
+  - Security Problems
+
+## Required.  Use Markdown formatting.  Only one paragraph.  No links allowed.
+## Should be less than 500 characters
+excerpt: >
+  **Transaction bloom filtering** is a method that allows lightweight
+  clients to limit the amount of transaction data they receive from full
+  nodes to only those transactions that affect their wallet (plus a
+  configurable amount of additional transactions to to generate
+  plausible denaibliity about which transactions belong to the client).
+
+## Optional.  Use Markdown formatting.  Multiple paragraphs.  Links allowed.
+extended_summary: |
+  [Bloom filters][] provide the ability to create a compact filter that
+  is guarantee to match a specified string with a configurable rate of
+  false positive matches for other strings.  A lightweight client can
+  create a bloom filter for all of its wallet addresses, send that
+  filter to a node using the P2P protocol messages defined in BIP37, and
+  then request a special form of blocks (merkle blocks) from the node.
+
+  A merkle block, also defined in BIP37, will contain only transactions
+  matching the previously sent filter plus the block header and a
+  partial merkle branch connecting each matching transaction to the
+  merkle root in the block header.
+
+  Clients will also receive announcements of new unconfirmed transactions being
+  relayed by the node if they match the filter.
+
+  When BIP37 was popular, most lightweight clients that used it ran on
+  mobile devices with limited bandwidth and so chose low false positive
+  rates to minimize their bandwidth use.  This meant that they
+  essentially gave their list of addresses to any node they contacted.
+  It was expected that privacy-focused users could mitigate this privacy
+  loss by setting a higher false positive rate, but research suggests
+  that the rate needs to be quite high in order to provide plausible
+  deniability.  The problem is a consequence of the different types of
```

Comment:
Is there a source we can link to that explains this? Everything from "The problem is a consequence of..." is perhaps a bit too low-level for a summary.

Reply:
Yeah, Jonas Nick's article already linked in the see also section.  https://jonasnick.github.io/blog/2015/02/12/privacy-in-bitcoinj/